docs/reference/go/go.md

```diff
@@ -0,0 +1,194 @@
+---
+bodyclass: docs
+headline: Go Generated Code Guide
+layout: docs
+type: markdown
+---
+
+<p class="lead">This guide gives an overview of code generated with the protoc grpc go plugin.</p>
+
+<div id="toc"></div>
+
+## Compiler Invocation and Results of Code Generation
+This section provides instructions for using the [grpc plugin](https://godoc.org/github.com/golang/protobuf/protoc-gen-go/grpc) to `protoc-gen-go`, when compiling `.proto` files with `protoc`.
+See the Protocol Buffers [Go Generated Code Guide](https://developers.google.com/protocol-buffers/docs/reference/go-generated) for detailed instructions on 
+how to use `protoc` compiler with the `protoc-gen-go` plugin.
+
+You can find out how to define a gRPC service in a .proto file in [Service Definitions](../guides/concepts.html#service-definition).
+
+### Different RPC types and their RPC signatures in proto files
+The different types RPC's (unary, client-streaming, server-streaming, and bidi-streaming),
+are declared in the `service` declarations in `.proto` files as follows:
+
+Note that type message types `MsgA` and `MsgB` are just placeholders for arbitrary protobuf message types. Also note that the descriptions
+further on this document refer to the example RPC signatures in this subsection.
+
+Thread-safety: note that client-side RPC invocations and server-side RPC handlers <i>are thread-safe</i> and are meant
```

Comment:
(I had to put the text in a code snippet so you could see the markup, obviously we just want it as HTML in the file!)